dbt/adapters/bigquery/dataset.py

```diff
@@ -20,15 +20,9 @@ def add_access_entry_to_dataset(dataset: Dataset, access_entry: AccessEntry) ->
     # we can't simply check if an access entry is in the list as the current equality check
     # does not work because the locally created AccessEntry can have extra properties.
     for existing_entry in access_entries:
-        role_match = existing_entry.role == access_entry.role
```

Comment:
It looks like we dropped the check on `.role` and `.entity_type`. Are those covered by checking `._properties`?

Reply:
turns out they are not covered by properties